src/rgw/rgw_lc.cc

```diff
@@ -351,13 +351,13 @@ int RGWLC::bucket_lc_process(string& shard_id)
   string bucket_id = result[2];
   int ret = store->get_bucket_info(obj_ctx, bucket_tenant, bucket_name, bucket_info, NULL, &bucket_attrs);
   if (ret < 0) {
-    ldout(cct, 0) << "LC:get_bucket_info failed" << bucket_name <<dendl;
+    ldout(cct, 0) << "LC:get_bucket_info for " << bucket_name << " failed" << dendl;
     return ret;
   }
 
   ret = bucket_info.bucket.bucket_id.compare(bucket_id) ;
-  if (ret !=0) {
-    ldout(cct, 0) << "LC:old bucket id find, should be delete" << bucket_name <<dendl;
+  if (ret != 0) {
+    ldout(cct, 0) << "LC:old bucket id found. " << bucket_name << " should be deleted" << dendl;
```

Comment:
This check should be `if (ret == 0)` in-order to find an old bucket id, right ?
@cbodley Could you please re-check this ?

Reply:
No, the check is ok.

The compare here is to verify the 'bucket_id' in debug.rgw.log/lc.<index> is valid.
Failed compare means the 'bucket_id' in debug.rgw.log/lc.<index> is invalid, when we shouldn't run lifecycle for this bucket.
One possible circumstance is the bucket whose id is 'bucket_id' was removed, after which a new bucket with the same name is created.

Perhaps this line is better:
ldout(cct, 0) << "LC:old bucket id found. " << bucket_name << " has been deleted and re-created" << dendl;

Reply:
@qrGitHub ok, thanks for clarifying.